test/functional/tests/security/test_load_corrupted_metadata.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# Copyright(c) 2023-2024 Huawei Technologies Co., Ltd.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+import random
+
+from api.cas import casadm
+from api.cas.cache_config import (
+    CacheMode,
+    CacheLineSize,
+    CleaningPolicy,
+    PromotionPolicy,
+    SeqCutOffPolicy,
+)
+from storage_devices.device import Device
+from core.test_run import TestRun
+from storage_devices.ramdisk import RamDisk
+from test_tools.dd import Dd
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.parametrizex("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrizex("promotion_policy", PromotionPolicy)
+@pytest.mark.parametrizex("seq_cutoff_policy", SeqCutOffPolicy)
+def test_load_corrupted_metadata(cache_mode, cache_line_size, cleaning_policy,
+                                 promotion_policy, seq_cutoff_policy):
+    """
+        title: Security test for loading cache with corrupted metadata.
+        description: |
+          Validate the ability of Open CAS to load cache instance and work if metadata is corrupted.
+        pass_criteria:
+          - If metadata is recognized as corrupted, load operation should be aborted.
+          - If not then executing I/O operations on CAS device should not result in kernel panic.
+    """
+
+    iteration_per_config = 1000
+
+    with TestRun.step("Prepare RAM devices for test."):
+        cache_dev, core_dev = RamDisk.create(disk_size=Size(100, Unit.MiB), disk_count=2)
+
+    for iteration in TestRun.iteration(
+            range(iteration_per_config),
+            f"Corrupt metadata {iteration_per_config} times."
+    ):
+
+        with TestRun.step("Prepare cache and core."):
+            cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+            core = casadm.add_core(cache, core_dev)
+            metadata_size = int(cache.metadata_size_on_disk.get_value())
+
+        with TestRun.step("Configure cache."):
+            cache.set_cleaning_policy(cleaning_policy)
+            cache.set_promotion_policy(promotion_policy)
+            cache.set_seq_cutoff_policy(seq_cutoff_policy)
+
+        with TestRun.step("Fill cache with random data."):
```

Comment:
```suggestion
        with TestRun.step("Fill cached volume with random data."):
```
As @Kamoppl pointed out, we cannot fill cache in PT ;) Nice catch!